pinot-core/src/main/java/org/apache/pinot/core/segment/virtualcolumn/SingleStringVirtualColumnProvider.java

```diff
@@ -52,8 +53,7 @@ public Dictionary buildDictionary(VirtualColumnContext context) {
   public ColumnMetadata buildMetadata(VirtualColumnContext context) {
     ColumnMetadata.Builder columnMetadataBuilder = super.getColumnMetadataBuilder(context);
 
-    columnMetadataBuilder.setCardinality(1).setHasDictionary(true).setHasInvertedIndex(true)
-        .setFieldType(FieldSpec.FieldType.DIMENSION).setDataType(FieldSpec.DataType.STRING).setSingleValue(true)
+    columnMetadataBuilder.setCardinality(1).setHasDictionary(true).setHasInvertedIndex(true).setSingleValue(true)
```

Comment:
Field type and data type is not required?

Reply:
Common metadata are moved to `BaseVirtualColumnProvider` (`setFieldType(fieldSpec.getFieldType()).setDataType(fieldSpec.getDataType()`)